build.gradle.kts

```diff
@@ -106,6 +106,34 @@ tasks {
         )
     }
 
+    val jvmTest = named<Test>("jvmTest") {
+        if (System.getProperty("idea.active") != "true") {
+            // We need to be able to run these tests in IntelliJ IDEA.
+            // Unfortunately, the current Gradle support doesn't detect
+            // the `jvmTestIsolated` task.
+            exclude("**/*IsolatedTest*")
+        }
+        configureJvmTestCommon()
+        val runAllTestsInSeparateJVMs: String by project
+        forkEvery = if (runAllTestsInSeparateJVMs.toBoolean()) 1 else 0
+    }
+
+    val jvmTestIsolated = register<Test>("jvmTestIsolated") {
+        group = jvmTest.get().group
+        testClassesDirs = jvmTest.get().testClassesDirs
+        classpath = jvmTest.get().classpath
+        enableAssertions = true
+        testLogging.showStandardStreams = true
+        outputs.upToDateWhen { false } // Always run tests when called
+        include("**/*IsolatedTest*")
```

Comment:
Just for the record. 
It seems there is a more robust mechanism to group JUnit tests using "test categories":

- https://github.com/junit-team/junit4/wiki/Categories
- https://docs.gradle.org/current/userguide/java_testing.html#test_grouping

Reply:
This is indeed a nice JUnit feature, thanks! However, in our case, Gradle will launch a new JVM instance for each thread, and then JUnit will decide whether to run the test or skip it, resulting in the build time increase by x1.5. We need to skip almost all tests in the `jvmTestIsolated` task. For that, the current solution is better.